internal/evaluation_target/application_snapshot_image/application_snapshot_image.go

```diff
@@ -163,13 +163,19 @@ func (a *ApplicationSnapshotImage) SetImageURL(url string) error {
 
 // ValidateImageSignature executes the cosign.VerifyImageSignature method on the ApplicationSnapshotImage image ref.
 func (a *ApplicationSnapshotImage) ValidateImageSignature(ctx context.Context) error {
-	_, _, err := NewClient(ctx).VerifyImageSignatures(ctx, a.reference, &a.checkOpts)
+	// Set the expected ClaimVerifier on a shallow *copy* of CheckOpts
+	opts := a.checkOpts
```

Comment:
Do we need to do this? If so perhaps the comment could elaborate as to why

Reply:
I think we do because otherwise, this method is causing a non-obvious side effect. I'll comment that.